### ISSUE TITLE
Create tags

### ISSUE THREAD

**Issue by elboletaire** [titpetric/netdata#28]
Having docker automated builds, creating tags in github should create the respective tag in docker hub.

Could it be possible to start creating tags from now?

We can even still add the old tags now, just checking the code in that version and building the image there. With a few lines of bash this could be easily done (I guess).

**Comment by titpetric** [comment]
I'll put it on the todo list, but it might be a while before I'm able to implement it fully. I hope it's not a high priority for you as it might well cross over into 2017.

**Comment by elboletaire** [comment]
Not a priority at all. Being a priority I'd help creating some bash script for the old tags, or maybe with some free time...

**Comment by elisiariocouto** [comment]
+1

**Comment by titpetric** [comment]
@elisiariocouto if it's a must-have for you (or anybody else), please don't add +1 to the comments and use reactions on the original issue post. Of course, you're welcome to submit a PR, currently this has no planned due date.

**Comment by elisiariocouto** [comment]
Alright, I can help you with this. How are the Docker Hub builds being triggered?

**Comment by titpetric** [comment]
Webhook, on commit of firehol/netdata (trigger via API) repository or on push to titpetric/netdata (standard). Nothing special about it. I already set it up once to use branches as tags (but never created appropriate branches for it):

![image](https://cloud.githubusercontent.com/assets/233360/22250508/77fed9d4-e247-11e6-9e64-c3976718a950.png)

So, technically, what needs to be done, ideally - is a script which I would put in the master branch, and occasionally run to update either tags or branches, which would then in turn use this information to pull the correct tag from firehol/netdata and build it as a docker container. The obvious issue might be, that the Dockerfile respectively changed over time and it might not be able to build whichever versions, meaning some modifications might have to be made per-branch.

Of course, we can also stick everything into a folder structure under the master branch, and I can create a few links on docker hub manually. It would mean however, that each new netdata version tag would need to be added manually to docker hub. I guess I am also fine with that.

**Comment by elisiariocouto** [comment]
Wouldn't this be much easier if the Docker Hub Automated builds were triggered by `firehol/netdata` repository itself? This way the only build setting that needs to be added is the `Tag` option.

<img width="1243" alt="screen shot 2017-01-24 at 15 05 03" src="https://cloud.githubusercontent.com/assets/818914/22252660/d33e2ada-e246-11e6-8da3-0d35a2b59347.png">

This way, everytime a new tag is pushed on `firehol/netdata` repository, a new Docker image is tagged with the same tag name. What do you think?

**Comment by titpetric** [comment]
Alas the netdata image is build from this repository. I don't think hub supports linking two different repositories like that. The only way I see is that the tag information would be transfered to this repository somehow (occasional commit to update tags).

**Comment by elboletaire** [comment]
If there's no way to do it directly with the hub we should consider using travis.

**Comment by elboletaire** [comment]
But it won't be autoexecuted on firehol's updates... so I'mnot really sure how to do this :confused:

**Comment by titpetric** [comment]
firehol/netdata has a custom webhook that triggers the titpetric/netdata hub build. I'm sure the @ktsaou will add a new one if needed.

**Comment by elisiariocouto** [comment]
Again, I think it is better if the `build.sh` and all the docker build related scripts are moved to `firehol/netdata` and a new Docker Hub account and Docker Automated Build is created by @ktsaou. This way, in just a few clicks, tags work and a proper Docker Hub repo is created. But that's just my opinion.

**Comment by ktsaou** [comment]
Guys, I am open to help in any way you like.
@titpetric if you want me to add another hook or link the main repo to docker, let me know.

**Comment by titpetric** [comment]
I created the structure and tags for 1.4 and 1.5 versions of netdata. I extended the README.md with information how to add new sections. Ofcourse, when doing all of this I realized that perhaps there's no need for automated builds for these tags, because they are not being actively developed. Most likely it would be enough to pass a TAG ENV to the Dockerfile, and have a local script to build and push older releases of netdata only once. I might change this at some later point.

Any change of `firehol/netdata` will most likely never trigger a change in the docker hub build, unless the tag was re-created upstream. I don't think that is being done. Most likely, a tag of 1.5.1 should be created for this purpose, and the tag 1.5.0 should not be touched.

@ktsaou if you're using semver to version netdata, I'd suggest tagging them with [major].[minor] along with the [major].[minor].[patch] which you already use. The [major].[minor] tag should be updated to point at the latest [major].[minor].[patch] version. Either that, or have versioned branches for development (ie, like you have for 1.3.0, but don't have for 1.4.0 and 1.5.0?). I asume that master/tags are the current practice of tagging releases, so I completely ignored this branch.

> Note: leaving this issue open for a while to gather come feedback/comments. Thank you all.

**Comment by ktsaou** [comment]
> I asume that master/tags are the current practice of tagging releases, so I completely ignored this branch.

you are right.

> if you're using semver to version netdata, I'd suggest tagging them with [major].[minor] along with the [major].[minor].[patch] which you already use. The [major].[minor] tag should be updated to point at the latest [major].[minor].[patch] version

ok, I will do.

**Comment by titpetric** [comment]
Closing issue, so far everything works as expected.

**Comment by elboletaire** [comment]
Niiice! 😄
